module/os/linux/zfs/arc_os.c

```diff
@@ -57,8 +57,21 @@
 #include <sys/trace_zfs.h>
 #include <sys/aggsum.h>
 
-int64_t last_free_memory;
-free_memory_reason_t last_free_reason;
+/*
+ * This is a limit on how many pages the ARC shrinker makes available for
+ * eviction in response to one page allocation attempt.  Note that in
+ * practice, the kernel's shrinker can ask us to evict up to about 4x this
+ * for one allocation attempt.
+ *
+ * The default limit of 10,000 (in practice, 160MB per allocation attempt)
+ * limits the amount of time spent attempting to reclaim ARC memory to around
+ * 100ms second per allocation attempt, even with a small average compressed
+ * block size of ~8KB.
```

Comment:
Mentioning that this math assumes 4k pages would make this more clear.  Particularly these days when larger page sizes are no longer that exotic.  `10,000 pages * 4K page * 4 attempts = 160M`.  Correct?

Reply:
It's actually `x = 10,000 pages * 4KB/page * 2` (at `priority=0`, the shrinker logic tries to evict double what we have) `+ x/2 + x/4 + x/8 + x/16 + ...` (for priorities 1,2,3,... that it normally does before getting to priority=0)